pandas/core/indexes/datetimes.py

```diff
@@ -139,9 +139,21 @@ def wrapper(self, other):
 
             if isinstance(other, Index):
                 o_mask = other.values.view('i8') == libts.iNaT
+            elif isinstance(other, ABCSeries):
+                try:
+                    # GH#19800 On 32-bit Windows builds this fails, but we can
```

Comment:
huh?

Reply:
The last few commits have been troubleshooting appveyor-only errors.  Based on this line in the [logs](https://ci.appveyor.com/project/pandas-dev/pandas/build/1.0.10653/job/723oitybccbcr7bb) I guessed it was a 32-bit build, but now I see a "64" in their so am back to no-clue:

```
[gw1] win32 -- Python 3.6.4 C:\Miniconda3_64\envs\pandas\python.exe
```

It looks like in this build `pandas.tests.frame.test_operators.TestDataFrameOperators.test_comparison_invalid` is trying to compare a `DatetimeIndex` with a `Series[int32]`.  In other builds the same test 
 compares with a `Series[int64]` and goes through OK.  The exact error varies depending on which commit we're looking at, but the pertinent one is a call to `other.values.view('i8')` which raises `ValueError` with a `Series[int32]`.

Note that `pd.date_range('2016-01-01', periods=3) == pd.Series([0, 1, 2], dtype='i4')` raises locally, so the appveyor-specific behavior is whatever is causing it to pass an int32 series when everything else passes an int64.

Reply:
this it a bug then, something is assuming int64 but getting int32, if you want to attach the last: 

```
In [6]: pd.Series([0, 1, 2], dtype='i4').view('i8')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-b2f0cc20755c> in <module>()
----> 1 pd.Series([0, 1, 2], dtype='i4').view('i8')

~/pandas/pandas/core/series.py in view(self, dtype)
    537 
    538     def view(self, dtype=None):
--> 539         return self._constructor(self._values.view(dtype),
    540                                  index=self.index).__finalize__(self)
    541 

ValueError: new type not compatible with array.

In [7]: pd.Series([0, 1, 2], dtype='i4').astype('i8', copy=False)
Out[7]: 
0    0
1    1
2    2
dtype: int64
```

I get why this is not allowed, but its a bit annoying.